modules/blazingsql/src/blazingcluster.ts

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {DataFrame, DataType, Series, TypeMap} from '@rapidsai/cudf';
+import {ChildProcess, fork} from 'child_process';
+
+import {UcpContext} from './addon';
+import {BlazingContext} from './blazingcontext';
+
+export const CREATE_BLAZING_CONTEXT  = 'createBlazingContext';
+export const BLAZING_CONTEXT_CREATED = 'blazingContextCreated';
+
+export const CREATE_TABLE  = 'createTable';
+export const TABLE_CREATED = 'tableCreated';
+
+export const DROP_TABLE    = 'dropTable';
+export const TABLE_DROPPED = 'tableDropped';
+
+export const RUN_QUERY = 'runQuery';
+export const QUERY_RAN = 'ranQuery';
+
+export const CONFIG_OPTIONS = {
+  PROTOCOL: 'UCX',
+};
+
+let ctxToken = 0;
+
+export class BlazingCluster {
+  private workers: ChildProcess[];
+  // @ts-ignore
+  // instantiated within init()
+  private blazingContext: BlazingContext;
+
+  /**
+   * Initializes and returns an instance of BlazingCluster.
+   *
+   * @param numWorkers the number of child processes to spawn
+   *
+   * @example
+   * ```typescript
+   * import {BlazingCluster} from '@rapidsai/blazingsql';
+   *
+   * await BlazingContext.init();
+   * ```
+   */
+  static async init(numWorkers = 1): Promise<BlazingCluster> {
```

Comment:
Will need to expose some more config options for users too -- ip address, ports list (or port range), network interface etc.

Reply:
yeah agreed, I think we should allow people to setup logging as you did when debugging.